fdbserver/BlobManager.actor.cpp

```diff
@@ -3498,6 +3498,24 @@ ACTOR Future<Void> loadBlobGranuleMergeBoundaries(Reference<BlobManagerData> bmD
 	return Void();
 }
 
+// Update BM epoc
+ACTOR Future<Void> updateBMEpoch(Reference<BlobManagerData> bmData, int64_t newEpoch) {
+	state Reference<ReadYourWritesTransaction> tr = makeReference<ReadYourWritesTransaction>(bmData->db);
+	loop {
+		tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+		tr->setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+		try {
+			tr->set(blobManagerEpochKey, blobManagerEpochValueFor(newEpoch));
```

Comment:
This should update to the maximum of the current epoch on the cluster and the desired new epoch, just to be safe. For example, if we provisioned the new cluster and the blob manager crash looped for some reason like an invalid blob configuration or something, the epoch could already be higher on this cluster than the desired value, and then it would go backwards.
Also, adding a comment on why this is necessary, and either a comment that this function should only be used during restore and/or enforcing that this can only be used during restore would be other good safety things to add.

Reply:
make sense.  fixed the code to skip the update if current epoch is larger. also added comments to explain the context